00001-enarx-sprint-framework/README.md

```diff
@@ -0,0 +1,542 @@
+
+### Daily Standup
+
+Daily standup is the team's daily check-in meeting, and when the **Triage**
+process is performed. It should provide an opportunity for individual
+contributors (and guests) to ask questions, elaborate on blockers, and ask for
+help if needed.
+
+As noted before, triage is the single most important component of this meeting,
+and should therefore be done first. The exact process of triage is detailed
+separately in the [Triage](#triage) section. Triage should not be strictly
+time-bound; if productive discussions are happening around a new issue or PR,
+those should be allowed to happen naturally.
+
+The decision to make triage a part of daily standup is deliberate: it allows
+issues and PRs filed by "drive-by" contributors (that is, people not part of
+the project that still wish to contribute) to receive human consideration after
+a maximum of 24 working hours -- a desirable property for open source.
+
+Following triage, the scrum master will pull up the sprint board for an informal
+board review. This does not need to go over every single issue; rather, it
+should be an opportunity for team members to point out highlights or interesting
+updates to their tasks, and request help on blockers or otherwise.
+
+Finally, the rest of the meeting should be dedicated to open-ended project
+discussion. If there are guests to the call, the team should welcome them and
+allow them to ask questions. If there are any other topics to cover, they should
+be brought up here.
+
+Standup should take no longer than a half-hour, and should be scheduled as
+follows:
+
+- 10-15 min: triage
+- 15-20 min: Board review/project discussion (however long is left after triage)
+
+### Retrospective
+
+The final meeting is **Retrospective**, which represents the end of a
+sprint. It is a time for the whole team to reflect on work accomplished, demo
+new things, and discuss how to approach tasks that didn't get completed.
+
+Most of these topics will take the form of discussion. The only significant
+process to detail here is the workflow for reviewing issues and PRs from the
+last sprint, which is what should happen first.
+
+![Sprint Retro diagram](./sprint-retro.svg)
+
+Any issues that don't have an associated PR are assumed to be "not started".
+However, any issue that made it into the sprint in the first place is probably
+still important, so it should be considered as such for the next sprint as well.
+Thus, it will be sent to the next Planning meeting as a Proposed issue (see
+[Planning](#planning) for more details).
+
+If it _does_ have an associated PR, the next question to ask is whether that
+associated PR was merged or not. If it was, then the issue is completed. If not,
+that means the PR is still in review and likely just needs a bit more review
+time. Thus, it should stay as-is in the Under Review column.
+
+Beyond that, the rest of the meeting should be dedicated to demos or other
+high-level discussion about work accomplished.
```

Comment:
Should demos be assigned as part of the sprint? Does the sprint take into account the time spent on creating a demo in addition to resolving issues? Do we care if we end up with zero demos or a very large number of demos for a sprint? Do we want to have bounds on this?

Reply:
Great question.

For the moment, I'd imagine we'd have to lump the cycles needed to produce a demo in with the overall task, or define making the demo as a separate task. In any case, this is something the team would have to discuss during the prioritization/bandwidth discussion part of sprint planning.

Reply:
> Should demos be assigned as part of the sprint? Does the sprint take into account the time spent on creating a demo in addition to resolving issues?

Normally, they should. Demos should really be just that, a demonstration of what has been accomplished during the sprint, so the whole team and stakeholders can come around what they have built and see progress. This is based on the idea that each sprint should deliver incremental value, and that every sprint as such always produces working code. We may be still a bit away from that, so i would loosen that requirement to just demonstrating what has been worked on and progress there, not adamantly requiring that it is a "usable increment".